tests/cedict-service/cedict.test.js

```diff
@@ -0,0 +1,279 @@
+/* eslint-env jest */
+import Cedict from '@lexisCs/cedict-service/cedict'
+import CedictPermanentStorage from '@lexisCs/cedict-service/cedict-permanent-storage.js'
+import CedictConfig from '@lexisCs/configurations/cedict.js'
+import CedictC1 from '@lexisCsCedict/cedict-v20191029-c001.json'
+import CedictC2 from '@lexisCsCedict/cedict-v20191029-c002.json'
+import CedictC3 from '@lexisCsCedict/cedict-v20191029-c003.json'
+import CedictC4 from '@lexisCsCedict/cedict-v20191029-c004.json'
+require('fake-indexeddb/auto')
+
+describe('CedictPermanentStorage class', () => {
+  /**
+   * Replaces an original loader with a stub that will load local test data.
+   *
+   * @param {string} url - A URL of a chunk to be fetched.
+   * @returns {Promise<object>} Returns a promise that is resolved with
+   *          an object containing dictionary data.
+   */
+  Cedict.prototype._loadJson = (url) => {
+    const resources = {
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c001.json': CedictC1,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c002.json': CedictC2,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c003.json': CedictC3,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c004.json': CedictC4
+    }
+    return Promise.resolve(resources[url])
+  }
+
+  let dictionary = new Map() // eslint-disable-line prefer-const
+  ;[CedictC1.entries, CedictC2.entries, CedictC3.entries, CedictC4.entries].flat()
+    .forEach(entry => dictionary.set(entry.index, entry))
+  const dictionaryEntriesTotal = 33
```

Comment:
is there a reason this isn't calculated from the test data?

Reply:
Oh, just missed that one. Great suggestion, will implement it!